code/modules/shuttle/marine_dropship.dm

```diff
@@ -976,48 +1000,128 @@
 	return ..()
 
 
-/obj/machinery/computer/shuttle/shuttle_control/ui_interact(mob/user)
-	if(!allowed(user))
-		to_chat(user, "<span class='warning'>Access Denied!</span>")
+/obj/machinery/computer/shuttle/shuttle_control/ui_interact(mob/user, datum/tgui/ui)
+	if(!(SSshuttle.getShuttle(shuttleId)))
+		RelinkShuttleId()
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "ShuttleControl")
+		ui.open()
+
+/obj/machinery/computer/shuttle/shuttle_control/ui_state(mob/user)
+	return GLOB.access_state
```

Comment:
Why making a new one? standard one should be enough?

Reply:
The ones i have seen doesn't have an access check, so i made a new one that does. Shuttle control computers are behind an ID lock after all.

Reply:
Also this can be used on other machines in the future that wanna check for their access as well.